src/AppInstallerCommonCore/Manifest/Manifest.cpp

```diff
@@ -6,20 +6,53 @@
 
 namespace AppInstaller::Manifest
 {
-    ManifestVer::ManifestVer(std::string version, bool fullValidation) : Version(std::move(version), ".")
+    ManifestVer::ManifestVer(std::string_view version, bool fullValidation)
     {
         bool validationSuccess = true;
 
-        if (m_parts.size() > 3)
+        // Separate the extensions out
+        size_t hyphenPos = version.find_first_of('-');
+        if (hyphenPos != std::string_view::npos)
+        {
+            // The first part is the main version
+            Assign(std::string{ version.substr(0, hyphenPos) }, ".");
+
+            // The second part is the extensions
+            hyphenPos += 1;
+            while (hyphenPos < version.length())
+            {
+                size_t newPos = version.find_first_of('-', hyphenPos);
+
+                size_t length = (newPos == std::string::npos ? version.length() : newPos) - hyphenPos;
+                m_extensions.emplace_back(std::string{ version.substr(hyphenPos, length) }, ".");
+
+                hyphenPos += length + 1;
+            }
+        }
+        else
+        {
+            Assign(std::string{ version }, ".");
+        }
+
+        if (m_parts.size() > 3 ||
+            (!m_extensions.empty() && fullValidation))
         {
             validationSuccess = false;
         }
         else
         {
             for (size_t i = 0; i < m_parts.size(); i++)
             {
-                if (!m_parts[i].Other.empty() &&
-                    (i < 2 || fullValidation))
+                if (!m_parts[i].Other.empty())
+                {
+                    validationSuccess = false;
+                    break;
+                }
+            }
+
+            for (const Version& ext : m_extensions)
+            {
+                if (ext.GetParts().empty() || ext.GetParts()[0].Integer != 0)

```

Comment:
>[0].Integer != 0 [](start = 60, length = 16)

Are we going to do anything about the remaining parts in extensions?

Reply:
What do you mean "do anything"?  Validate?  I don't see any value to doing so.  Again, happy to be shown a reason and change it.

---
In reply to: [490692976](https://github.com/microsoft/winget-cli/pull/574#discussion_r490692976) [](ancestors = 490692976)